addons/website_slides/static/src/js/slides_course_quiz.js

```diff
@@ -261,13 +375,147 @@ odoo.define('website_slides.quiz', function (require) {
                 values.push(parseInt($(inputs[i]).val()));
             }
 
-            if (values.length === this.quiz.questions.length){
+            if (values.length === this.quiz.questionsCount){
                 this._alertHide();
                 this._submitQuiz(values);
             } else {
                 this._alertShow();
             }
         },
+
+        _onCreateQuizClick: function () {
+            var $elem = this.$('.o_wslides_js_quiz_add');
+            new QuestionFormWidget(this, {
+                slideId: this.slide.id,
+                sequence: this.quiz.questionsCount + 1
+            }).replace($elem);
+        },
+
+        _onEditQuestionClick: function (ev) {
+            var $elem = $(ev.currentTarget).closest('.o_wslides_js_lesson_quiz_question');
+            var question = this._getQuestionDetails($elem);
+            new QuestionFormWidget(this, {
+                oldElement: $elem,
+                question: question,
+                slideId: this.slide.id,
+                sequence: question.sequence,
+                update: true
+            }).replace($elem);
+        },
+
+        _onDeleteQuestionClick: function (ev) {
+            var question = $(ev.currentTarget).closest('.o_wslides_js_lesson_quiz_question');
+            new ConfirmationDialog(this, {
+                questionId: question.data('questionId'),
+                questionTitle: question.data('title')
+            }).open();
+        },
+
+        /**
+         * Displays the created Question at the correct place (after the last question or
+         * at the first place if there is no questions yet) It also displays the 'Add Question'
+         * button or open a new QuestionFormWidget if the user wants to immediately add another one.
+         *
+         * @param response
+         * @private
+         */
+        _displayCreatedQuestion: function (event) {
+            var $lastQuestion = this.$('.o_wslides_js_lesson_quiz_question:last');
+            if ($lastQuestion.length !== 0) {
+                $lastQuestion.after(event.data.view);
+            } else {
+                this.$el.prepend(event.data.view);
+            }
+            this.quiz.questionsCount++;
+            var $elem = this.$('.o_wsildes_quiz_question_input:last');
+            if (event.data.createNext) {
+                new QuestionFormWidget(this, {
+                    slideId: this.slide.id,
+                    sequence: this.quiz.questionsCount + 1
+                }).replace($elem);
+            } else {
+                $elem.replaceWith(QWeb.render('slide.quiz.new.question.button'));
+            }
```

Comment:
Not sure why you need to replace the last input thing.
Can you explain a little?

Reply:
the last selector is there when your are currently updating an existing question (which has also the .o_wsildes_quiz_question_input class) while creating a new one. It avoids to add the button on an wrong element.